src/tooltip.ts

```diff
@@ -0,0 +1,128 @@
+import { select, selectAll } from "d3";
+import { format as SSF } from "ssf";
+
+const yPadding = 15;
+const xPadding = 5;
+
+export function initTooltip(fontSize) {
+  selectAll("#tooltip").remove();
+  select("#vis")
+    .insert("div", ":first-child")
+    .attr("id", "tooltip")
+    .style("opacity", 0)
```

Comment:
My advice here would be to let D3 control the DOM but CSS control the styles -- it's _much_ faster . You can move all of the `.style()`s to a styled parent... probably in the `create` function... And then use a `.classed("tooltip", true") ` here to cut down. *totally* optional

Reply:
That's a great idea and there are several places where this could improve performance. Especially in the "on focus" feature -- i'll try to move these over to some css styles 👍